Formula/s/spidermonkey.rb

```diff
@@ -25,53 +24,70 @@ class Spidermonkey < Formula
+    depends_on "llvm" => :build # for lld
+  end
+
+  on_linux do
+    depends_on "icu4c"
+    depends_on "nspr"
+  end
+
   conflicts_with "narwhal", because: "both install a js binary"
 
   # From python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py
   fails_with :gcc do
-    version "6"
-    cause "Only GCC 7.1 or newer is supported"
+    version "7"
+    cause "Only GCC 8.1 or newer is supported"
   end
 
-  def install
-    # Avoid installing into HOMEBREW_PREFIX.
-    # https://github.com/Homebrew/homebrew-core/pull/98809
-    ENV["SETUPTOOLS_USE_DISTUTILS"] = "stdlib"
-
-    # Remove the broken *(for anyone but FF) install_name
-    # _LOADER_PATH := @executable_path
-    inreplace "config/rules.mk",
-              "-install_name $(_LOADER_PATH)/$(SHARED_LIBRARY) ",
-              "-install_name #{lib}/$(SHARED_LIBRARY) "
-
-    inreplace "old-configure", "-Wl,-executable_path,${DIST}/bin", ""
+  # Apply patch used by `gjs` to bypass build error.
+  # ERROR: *** The pkg-config script could not be found. Make sure it is
+  # *** in your path, or set the PKG_CONFIG environment variable
+  # *** to the full path to pkg-config.
+  # Ref: https://bugzilla.mozilla.org/show_bug.cgi?id=1783570
+  # Ref: https://discourse.gnome.org/t/gnome-45-to-depend-on-spidermonkey-115/16653
+  patch do
+    on_macos do
+      url "https://github.com/ptomato/mozjs/commit/9f778cec201f87fd68dc98380ac1097b2ff371e4.patch?full_index=1"
+      sha256 "a772f39e5370d263fd7e182effb1b2b990cae8c63783f5a6673f16737ff91573"
+    end
+  end
 
-    cd "js/src"
-    system "autoconf213"
+  def install
     mkdir "brew-build" do
-      system "../configure", "--prefix=#{prefix}",
-                             "--enable-optimize",
-                             "--enable-readline",
-                             "--enable-release",
-                             "--enable-shared-js",
-                             "--disable-bootstrap",
-                             "--disable-jemalloc",
-                             "--with-intl-api",
-                             "--with-system-icu",
-                             "--with-system-nspr",
-                             "--with-system-zlib"
+      args = %W[
+        --prefix=#{prefix}
+        --enable-optimize
+        --enable-readline
+        --enable-release
+        --enable-shared-js
+        --disable-bootstrap
+        --disable-debug
+        --disable-jemalloc
+        --with-intl-api
+        --with-system-zlib
+      ]
+      if OS.linux?
+        # System libraries are only supported on Linux and build fails if args are used on macOS.
+        # Ref: https://bugzilla.mozilla.org/show_bug.cgi?id=1776255
+        args += %w[--with-system-icu --with-system-nspr]
+      elsif MacOS.version >= :sonoma
+        # Use LLVM's lld to work around build script linker detection not working with newer Xcode
+        args << "--enable-linker=lld"
+        ENV.O3 # ld64.lld: error: -Os: number expected, but got 's'
```

Comment:
I think Mozilla's build scripts use some odd hack to detect Xcode ld64 that no longer works on Sonoma.

At least with newer `spidermonkey` a work around is `lld` but older `@91` will fail on our shims.

Reply:
Might be able to use `-Wl,-ld_classic` instead as upstream is expecting logging output when running failed command `-Wl,-version` with `LD_PRINT_OPTIONS=1`